avocado/utils/dmesg.py

```diff
@@ -0,0 +1,115 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import time
+import os
+import tempfile
+import logging
+
+from . import process, genio
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+class DmesgError(Exception):
+    """Base Exception Class for all dmesg  utils exceptions"""
+    pass
+
+
+def clear_dmesg():
+    """This function clear dmesg of system and save older dmesg in a file"""
+
+    cmd = "dmesg -c"
+    status = process.system(cmd, timeout=30, ignore_status=True,
+                            verbose=False, shell=True, sudo=True)
+    if status:
+        raise DmesgError(
+            "Unable to clear dmesg as some issue while clearing")
+
+
+def collect_dmesg(output_file=None):
+    """This Function collect dmesg and save in a tmp file
+
```

Comment:
Would be nice to improve the docstring here too and explain what is `output_file` since you are saying that saver in a tmp file.

Reply:
sure